src/ledger/insert_shred.zig

```diff
@@ -1820,6 +1893,7 @@ pub const BlockstoreInsertionMetrics = struct {
     }
 };
 
+/// Question: Is it possible to use WorkingEntry(SlotMeta) here?
```

Comment:
It's possible but would represent a different usage pattern. Normally WorkingEntry is not mutated and just represents whether the item is new or existing. It could probably be adapted for this use case though, where it needs to allow mutations, and it flips from clean to dirty as soon as a shred is inserted. I can't be sure which way is better without seeing it implemented.